modelforge/tests/test_dimenet.py

```diff
@@ -0,0 +1,124 @@
+from typing import Optional
+import pytest
+
+
+@pytest.fixture(scope="session")
+def prep_temp_dir(tmp_path_factory):
+    fn = tmp_path_factory.mktemp("test_dimenet_temp")
+    return fn
+
+
+def setup_dimenet_model(potential_seed: Optional[int] = None):
+    from modelforge.tests.test_potentials import load_configs_into_pydantic_models
+    from modelforge.potential import NeuralNetworkPotentialFactory
+
+    # read default parameters
+    config = load_configs_into_pydantic_models("dimenet", "qm9")
+
+    model = NeuralNetworkPotentialFactory.generate_potential(
+        use="inference",
+        potential_parameter=config["potential"],
+        training_parameter=config["training"],
+        dataset_parameter=config["dataset"],
+        runtime_parameter=config["runtime"],
+        potential_seed=potential_seed,
+        use_training_mode_neighborlist=True,
+        jit=False,
+    )
+    return model
+
+
+def test_init():
+    """Test initialization of the Dimenet model."""
+    potential = setup_dimenet_model()
+    assert potential is not None, "Dimenet model should be initialized."
+
+
+def test_forward(single_batch_with_batchsize, prep_temp_dir):
+    import torch
+
+    potential = setup_dimenet_model()
+    print(potential)
+
+    batch = single_batch_with_batchsize(
+        batch_size=64, dataset_name="QM9", local_cache_dir=str(prep_temp_dir)
+    )
+
+    yhat = potential(batch.nnp_input.to_dtype(dtype=torch.float32))
+
+
+def test_envelope():
+    from modelforge.potential.dimenet import Envelope
+    import torch
+
+    # Create an instance of the Envelope class
+    envelope = Envelope(exponent=5)
+
+    # Sample input tensor
+    inputs = torch.tensor([0.5, 0.8, 1.0, 1.2], dtype=torch.float32)
+
+    # Forward pass
+    outputs = envelope(inputs)
+    assert outputs.shape == inputs.shape
+    assert torch.allclose(outputs, torch.tensor([1.7109, 0.2539, 0.0000, 0.0000]), rtol=1e-3)
+
+    # Script the model for optimization and deployment
+    scripted_envelope = torch.jit.script(envelope)
+
+    # Verify that the scripted model works
+    outputs_scripted = scripted_envelope(inputs)
+    print(outputs_scripted)
+    assert torch.allclose(outputs, outputs_scripted)
+
+    # test for correct output computation
+    exponent=6 # Envelop function receives exponent = 5 but takes its increment and uses exponent = 6 -> why?
+    d_ij=0.5
```

Comment:
@mwieder, please review!

Reply:
we have resolved this! the exponent is now not incremented but is used as provided